content/posts/2025-01-30-ebpf-flows-filtering.md

```diff
@@ -0,0 +1,268 @@
+this flow should be allowed with a sampling rate of `30`.
+Reject all other traffic `default action`
+
+```yaml
+agent:
+  type: eBPF
+  ebpf:
+    flowFilter:
+      enable: true
+      rules:
+        - action: Accept
+          cidr: 172.30.0.0/16
+          peerCIDR: 10.128.0.0/14
+          sampling: 10
+        - action: Accept
+          cidr: 10.128.0.0/14
+          peerCIDR: 10.128.0.0/14
+          sampling: 20
+        - action: Accept
+          cidr: 8.8.8.8/32
+          sampling: 30
+          peerCIDR: 10.128.0.0/14
+          protocol: ICMP
+          icmpType: 8
+```
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-svc-and-pods-flows.png')}" alt="eBPF Flows Filtering Kubernetes NorthSouth and EastWest Flows">
+<p style="text-align: center">Figure 2: eBPF Flows Filtering Kubernetes NorthSouth and EastWest Flows</p>
+
+### Filter flows with packet drops
+
+Let's filter Kubernetes service flows that include a packet drop and discard all others.
+For this use case, the `PacketDrop` feature must be enabled with eBPF in `privileged` mode,
+as demonstrated in the configuration below.
+
+```yaml
+agent:
+  type: eBPF
+  ebpf:
+    privileged: true
+    features:
+      - PacketDrop
+    flowFilter:
+      enable: true
+      rules:
+        - action: Accept
+          cidr: 172.30.0.0/16
+          pktDrops: true
+        - action: Reject
+          cidr: 0.0.0.0/0
+```
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-filter-svc-pkt-drops.png')}" alt="eBPF Flows Filtering Kubernetes Services with Packet Drop">
+<p style="text-align: center">Figure 3: eBPF Flows Filtering Kubernetes Services with Packet Drop</p>
+
+### Filter TCP flows using TCP Flags
+
+Filtering based on TCP flags is an effective method to detect and mitigate TCP SYN flood attacks in a cluster.
+A SYN flood is a Denial-of-Service (DoS) attack where an attacker overwhelms a target system by sending a large
+number of SYN packets without completing the three-way handshake,
+depleting system resources and disrupting legitimate connections.
+
+```yaml
+agent:
+  type: eBPF
+  ebpf:
+    flowFilter:
+      enable: true
+      rules:
+        - action: Accept
+          cidr: 0.0.0.0/0
+          protocol: TCP
+          tcpFlags: SYN
+          sampling: 1
+```
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-filter-with-tcpflags.png')}" alt="eBPF Flows Filtering TCP flows using TCP flags">
+<p style="text-align: center">Figure 4: eBPF Flows Filtering TCP flows using TCP flags</p>
+
+### Filter DNS query over ports 53 and 5353 for both TCP and UDP
+
+This Use case involves capturing DNS flows over both `TCP` and `UDP`,
+with the option to enable the `DNSTracking` feature for enhanced DNS latency insights.
+
+```yaml
+agent:
+  type: eBPF
+  ebpf:
+    features:
+      - DNSTracking
+    flowFilter:
+      enable: true
+      rules:
+        - action: Accept
+          cidr: 0.0.0.0/0
+          sourcePorts: '53,5353'
+          sampling: 1
```

Comment:
For this to work properly, don't you need to set DNS_TRACKING_PORT?
```
    advanced:
      env:
        DNS_TRACKING_PORT: "5353"```

Reply:
I didn't want to export this back door config in the blog but ofc for that u need DNS to listen to 5353

Reply:
ok /lgtm